packages/ozone/src/api/admin/emitModerationEvent.ts

```diff
@@ -2,156 +2,204 @@ import { AuthRequiredError, InvalidRequestError } from '@atproto/xrpc-server'
+        createdBy: ctx.cfg.service.did,
+        subjectStatus: result.subjectStatus,
+      })
 
-        if (subject.isRepo()) {
-          if (isTakedownEvent) {
-            const isSuspend = !!result.event.durationInHours
-            await moderationTxn.takedownRepo(
-              subject,
-              result.event.id,
-              isSuspend,
-            )
-          } else if (isReverseTakedownEvent) {
-            await moderationTxn.reverseTakedownRepo(subject)
-          }
+      if (subject.isRepo()) {
+        if (isTakedownEvent) {
+          const isSuspend = !!result.event.durationInHours
+          await moderationTxn.takedownRepo(subject, result.event.id, isSuspend)
+        } else if (isReverseTakedownEvent) {
+          await moderationTxn.reverseTakedownRepo(subject)
         }
+      }
 
-        if (subject.isRecord()) {
-          if (isTakedownEvent) {
-            await moderationTxn.takedownRecord(subject, result.event.id)
-          } else if (isReverseTakedownEvent) {
-            await moderationTxn.reverseTakedownRecord(subject)
-          }
+      if (subject.isRecord()) {
+        if (isTakedownEvent) {
+          await moderationTxn.takedownRecord(subject, result.event.id)
+        } else if (isReverseTakedownEvent) {
+          await moderationTxn.reverseTakedownRecord(subject)
         }
+      }
 
-        if (isLabelEvent) {
-          await moderationTxn.formatAndCreateLabels(
-            result.event.subjectUri ?? result.event.subjectDid,
-            result.event.subjectCid,
-            {
-              create: result.event.createLabelVals?.length
-                ? result.event.createLabelVals.split(' ')
-                : undefined,
-              negate: result.event.negateLabelVals?.length
-                ? result.event.negateLabelVals.split(' ')
-                : undefined,
-            },
-          )
-        }
+      if (isLabelEvent) {
+        await moderationTxn.formatAndCreateLabels(
+          result.event.subjectUri ?? result.event.subjectDid,
+          result.event.subjectCid,
+          {
+            create: result.event.createLabelVals?.length
+              ? result.event.createLabelVals.split(' ')
+              : undefined,
+            negate: result.event.negateLabelVals?.length
+              ? result.event.negateLabelVals.split(' ')
+              : undefined,
+          },
+        )
+      }
 
-        return result.event
+      return result.event
+    })
+
+    return moderationService.views.formatEvent(moderationEvent)
+  }
+}
+
+export default function (server: Server, ctx: AppContext) {
+  server.com.atproto.admin.emitModerationEvent({
+    auth: ctx.authVerifier.modOrRole,
+    handler: async ({ input, auth }) => {
+      const moderationEvent = await handleModerationEvent({
+        input,
+        auth,
+        ctx,
       })
 
+      // On divert events, we need to automatically take down the blobs
+      if (isModEventDivert(input.body.event)) {
```

Comment:
emit takedown event following divert event.

Reply:
One event expanding into two events on the backend is a little funky.  If we find ourselves needing to do this again it might be worth revisiting, seeing if there's a nice way to think about this.